cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381/Internal.hs

```diff
@@ -288,6 +318,40 @@ withNewAffine_ = fmap fst . withNewAffine
 withNewAffine' :: BLS curve => (AffinePtr curve -> IO a) -> IO (Affine curve)
 withNewAffine' = fmap snd . withNewAffine
 
+withPointArray :: [Point curve] -> (PointArrayPtr curve -> IO a) -> IO a
```

Comment:
Relying on null-termination is dangerous. Right now, usages of this function are safe, but this is brittle: the second argument to `withPointArray` there closes over the size of the original list, so it gets passed in "out of band", and implicitly depends on `withPointArray` keeping the original list length unchanged (which right now it does, but that's an implicit assumption, not part of the explicit API).

`withPointArray` as it is invites incorrect usage, so I would suggest passing the size explicitly (cf. `CString` vs. `CStringLen`):

```suggestion
withPointArray :: [Point curve] -> (Int -> PointArrayPtr curve -> IO a) -> IO a
```